modules/custom/activity_viewer/src/Plugin/views/filter/ActivityFilterPersonalisedHomepage.php

```diff
@@ -77,163 +108,259 @@ public function canExpose() {
    */
   public function query() {
     $account = $this->view->getUser();
-    $group_memberships = $this->groupHelper->getAllGroupsForUser($account->id());
-
-    // Add tables and joins.
-    $this->query->addTable('activity__field_activity_recipient_group');
-    $this->query->addTable('activity__field_activity_entity');
-
-    $configuration = [
-      'left_table' => 'activity__field_activity_entity',
-      'left_field' => 'field_activity_entity_target_id',
-      'table' => 'post_field_data',
-      'field' => 'id',
-      'operator' => '=',
-      'extra' => [
-        0 => [
-          'left_field' => 'field_activity_entity_target_type',
-          'value' => 'post',
-        ],
-      ],
-    ];
-    $join = Views::pluginManager('join')->createInstance('standard', $configuration);
-    $this->query->addRelationship('post', $join, 'activity__field_activity_entity');
-
-    $configuration = [
-      'left_table' => 'post',
-      'left_field' => 'id',
-      'table' => 'post__field_visibility',
-      'field' => 'entity_id',
-      'operator' => '=',
-    ];
-    $join = Views::pluginManager('join')->createInstance('standard', $configuration);
-    $this->query->addRelationship('post__field_visibility', $join, 'post__field_visibility');
-
-    // Join node table(s).
-    $configuration = [
-      'left_table' => 'activity__field_activity_entity',
-      'left_field' => 'field_activity_entity_target_id',
-      'table' => 'node_access',
-      'field' => 'nid',
-      'operator' => '=',
-      'extra' => [
-        0 => [
-          'left_field' => 'field_activity_entity_target_type',
-          'value' => 'node',
-        ],
-      ],
+    $skip_roles = [
+      'administrator',
+      'contentmanager',
+      'sitemanager',
     ];
-    $join = Views::pluginManager('join')->createInstance('standard', $configuration);
-    $this->query->addRelationship('node_access', $join, 'node_access_relationship');
+    $hide_from_view = $nids = $pids = $cids = [];
 
-    if ($account->isAnonymous()) {
-      $configuration['table'] = 'node_field_data';
-      $join = Views::pluginManager('join')->createInstance('standard', $configuration);
-      $this->query->addRelationship('node_field_data', $join, 'node_field_data');
+    // Skip filter for users that have full access to the site content.
+    if (!empty(array_intersect($skip_roles, $account->getRoles()))) {
+      return;
     }
 
+    $this->ensureMyTable();
+    $group_memberships = $this->groupHelper->getAllGroupsForUser($account->id());
+    $this->query->addTable('activity__field_activity_entity');
```

Comment:
@dmytriikaiun Could you check those notices from PHPStan?

Reply:
@ribel Resolved